native/explorer/src/dataframe/io.rs

```diff
@@ -230,6 +231,59 @@ pub fn df_to_parquet(
     Ok(())
 }
 
+#[cfg(feature = "aws")]
+#[rustler::nif(schedule = "DirtyIo")]
+pub fn df_to_parquet_cloud(
+    data: ExDataFrame,
+    ex_entry: ExS3Entry,
+    ex_compression: ExParquetCompression,
+) -> Result<(), ExplorerError> {
+    use object_store::{aws::AmazonS3Builder, ObjectStore};
+    let config = ex_entry.config;
+    let mut aws_builder = AmazonS3Builder::new()
+        .with_region(config.region)
+        .with_bucket_name(ex_entry.bucket)
+        .with_access_key_id(config.access_key_id)
+        .with_secret_access_key(config.secret_access_key);
+
+    if let Some(endpoint) = config.endpoint {
+        aws_builder = aws_builder.with_allow_http(true).with_endpoint(endpoint);
+    }
+
+    if let Some(token) = config.token {
+        aws_builder = aws_builder.with_token(token);
+    }
+
+    let aws = aws_builder
+        .build()
+        .map_err(object_store_to_explorer_error)?;
+
+    let object_store: Box<dyn ObjectStore> = Box::new(aws);
+    let mut cloud_writer = crate::cloud_writer::CloudWriter::new(object_store, ex_entry.key.into());
+
+    let compression = ParquetCompression::try_from(ex_compression)?;
+
+    ParquetWriter::new(&mut cloud_writer)
+        .with_compression(compression)
+        .finish(&mut data.clone())?;
+    Ok(())
+}
+fn object_store_to_explorer_error(error: impl std::fmt::Debug) -> ExplorerError {
+    ExplorerError::Other(format!("Internal Object Store error: #{error:?}"))
+}
+
+#[cfg(not(feature = "aws"))]
+#[rustler::nif]
+pub fn df_to_parquet_cloud(
+    _data: ExDataFrame,
+    _ex_entry: ExS3Entry,
+    _ex_compression: ExParquetCompression,
+) -> Result<(), ExplorerError> {
+    Err(ExplorerError::Other(format!(
+        "AWS reads and writes are not enabled for this machine"
```

Comment:
Not sure what we do in other cases, but maybe it is helpful to mention that Explorer itself is compiled without support for it on this machine? The current message seems rather vague to me, i.e. difficult to search for online, and not immediately clear from seeing the message what the cause is and how to fix it.

Reply:
Makes sense! I'm going to improve the messages related to this.